### DISCUSSION TITLE
How is the mean and s.e. calculated in partial plots?

### DISCUSSION THREAD

**Discussion by mao223** [kogalur/randomForestSRC#272] [Q&A] [answered]
For the quantity:
```
plot.variable(airq.obj, xvar.names = "Wind", partial = TRUE)$pData[[1]]$yhat
```
Is it correct that these are the means over the rows (for each unique value for Wind set)? Is it possible to return the raw predictions per row?

How is the quantity below calculated?
```
plot.variable(airq.obj, xvar.names = "Wind", partial = TRUE)$pData[[1]]$yhat.se
```
Is it simply s.d. / sqrt(n)? Is there bootstrapping involved?

Thank you!

**Comment by ishwaran** [comment, marked as answer]
Yes, the partial plot values are essentially an average where the averaging is over all other variables except the variable(s) being studied (in your example, "Wind" is the variable of interest).  If you are interested, a more detailed description is provided in the [partial plot vignette](https://www.randomforestsrc.org/articles/partial.html).

Yes, it is possible to return the raw values.  You should use the function `partial` which is the fast and direct interface for constructing partial plot values (the `plot.variable` function is simply a user friendly wrapper to the primary function `partial`).

In your example, this would look like:

```
airq.obj <- rfsrc(Ozone ~ ., data = airquality)
partial.obj <- partial(airq.obj,
                       partial.xvar = "Wind",
                       partial.values = airq.obj$xvar$Wind)
```

The object `partial.obj` contains a lot of information and can be overwhelming but there is a simple function for extracting this information `get.partial.plot.data`

`pdta <- get.partial.plot.data(partial.obj)`

In this example this function is processing the information in `partial.obj$regrOutput[[1]]` which is a matrix where the rows are the data points and the columns are the unique values of Wind.  So for example, if you look at a specific column this gives you all the partial plot values over the data for that unique value of Wind.  This gives you what you want.

In fact, you will find that

 `colMeans(partial.obj$regrOutput[[1]])`

is the same as `pdta`

Finally, there is no bootstrapping being used for the so-called standard error.  It is simply s.d/sqrt(n) as you surmised.  This is very crude and should be treated with caution.

**Reply by mao223** [reply]
Thank you so much Dr. Ishwaran for your help
